modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/registries/Embedded.scala

```diff
@@ -0,0 +1,151 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.iglu.client.resolver.registries
+
+// Java
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaVer}
+
+import java.io.{File, InputStream}
+import scala.util.matching.Regex
+
+// Scala
+import scala.io.Source
+import scala.util.control.NonFatal
+
+// Cats
+import cats.effect.Sync
+import cats.effect.implicits._
+import cats.implicits._
+import cats.data.EitherT
+
+// circe
+import io.circe.parser.parse
+import io.circe.Json
+
+import com.snowplowanalytics.iglu.core.SchemaList
+
+private[registries] object Embedded {
+
+  /**
+   * Retrieves an Iglu Schema from the Embedded Iglu Repo as a JSON
+   *
+   * @param base path on the local filesystem system
+   * @param key The SchemaKey uniquely identifying the schema in Iglu
+   * @return either a `Json` on success, or `RegistryError` in case of any failure
+   *         (i.e. all exceptions should be swallowed by `RegistryError`)
+   */
+  def lookup[F[_]: Sync](
+    base: String,
+    key: SchemaKey
+  ): F[Either[RegistryError, Json]] = {
+    val path   = RegistryLookup.toPath(base, key)
+    val is     = readResource[F](path)
+    val schema = is.bracket(_.traverse(fromStream[F]))(_.traverse_(closeStream[F]))
+    val result = for {
+      stringOption <- schema.attemptT.leftMap(Utils.repoFailure)
+      string       <- EitherT.fromOption[F](stringOption, RegistryError.NotFound: RegistryError)
+      json         <- EitherT.fromEither[F](parse(string)).leftMap(Utils.invalidSchema)
+    } yield json
+
+    result.value
+  }
+
+  /** Not-RT analog of [[Embedded.lookup]] */
+  def unsafeLookup(path: String): Either[RegistryError, Json] =
+    try {
+      val is     = unsafeReadResource(path)
+      val schema = is.map(unsafeFromStream)
+      val result = schema
+        .toRight(RegistryError.NotFound: RegistryError)
+        .flatMap(x => parse(x).leftMap(Utils.invalidSchema))
+      is.fold(())(unsafeCloseStream)
+      result
+    } catch {
+      case NonFatal(e) =>
+        e match {
+          case _: NullPointerException => RegistryError.NotFound.asLeft
+          case _                       => Utils.repoFailure(e).asLeft
+        }
+    }
+
+  def unsafeList(path: String, modelMatch: Int): Either[RegistryError, SchemaList] =
+    try {
+      val d =
+        new File(
+          getClass.getResource(path).getPath
+        ) // this will throw NPE for missing entry in embedded repos
+      val schemaFileRegex: Regex = (".*/schemas/?" + // path to file
+        "([a-zA-Z0-9-_.]+)/" +            // Vendor
+        "([a-zA-Z0-9-_]+)/" +             // Name
+        "([a-zA-Z0-9-_]+)/" +             // Format
+        "([1-9][0-9]*)-(\\d+)-(\\d+)$").r // MODEL, REVISION and ADDITION
```

Comment:
we also repeat this regex across many places, feels like this regex should also come from the core lib (where it is tested) not to cause a potential bug